File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/Max/MaxRewardVideoVC.m
#import "MaxRewardVideoVC.h"
#import <AppLovinSDK/AppLovinSDK.h>
#import "AdConfig.h"

@interface MaxRewardVideoVC () <MARewardedAdDelegate>

@property (nonatomic, strong) UILabel *label;
@property (nonatomic, assign) BOOL isLoading;
@property (nonatomic, strong, nullable) MARewardedAd *rewardAd;
@property (nonatomic, copy) NSString *TAG;

@end

@implementation MaxRewardVideoVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.TAG = @"Max-rewardVideo:";
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationItem.title = NSLocalizedString(@"max_rewardVideo", @"");

    UIButton *bnLoad = [self createButtonWithTitle:NSLocalizedString(@"load_ad", @"") action:@selector(buttonLoad)];
    [self.view addSubview:bnLoad];

    UIButton *bnShow = [self createButtonWithTitle:NSLocalizedString(@"show_ad", @"") action:@selector(buttonShow)];
    [self.view addSubview:bnShow];

    self.label = [self createLabel];
    [self.view addSubview:self.label];

    [NSLayoutConstraint activateConstraints:@[
        [bnLoad.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnLoad.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnLoad.heightAnchor constraintEqualToConstant:50],
        [bnLoad.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],

        [bnShow.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnShow.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnShow.heightAnchor constraintEqualToConstant:50],
        [bnShow.topAnchor constraintEqualToAnchor:bnLoad.bottomAnchor constant:20],

        [self.label.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.label.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.label.heightAnchor constraintEqualToConstant:50],
        [self.label.topAnchor constraintEqualToAnchor:bnShow.bottomAnchor constant:20],
    ]];
}

- (void)buttonLoad {
    NSLog(@"bnLoad");
    if (self.isLoading) {
        return;
    }
    [self updateUI:YES message:NSLocalizedString(@"loading", @"")];
    [self loadAd];
}

- (void)buttonShow {
    if (self.rewardAd && [self.rewardAd isReady]) {
        [self.rewardAd showAd];
    } else {
        NSLog(@"Ad wasn't ready");
    }
}

- (void)loadAd {
    self.rewardAd = [MARewardedAd sharedWithAdUnitIdentifier:[AdConfig Max_Reward_Video_Ad_Id]];
    self.rewardAd.delegate = self;
    [self.rewardAd loadAd];
}

- (void)updateUI:(BOOL)loading message:(NSString *)msg {
    self.isLoading = loading;
    self.label.text = msg;
}

#pragma mark - MARewardedAdDelegate

- (void)didLoadAd:(MAAd *)ad {
    NSLog(@"%@ didLoad", self.TAG);
    [self updateUI:NO message:NSLocalizedString(@"load_success", @"")];
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(MAError *)error {
    NSString *msg = [NSString stringWithFormat:@"%ld: %@", (long)error.code, error.description];
    NSLog(@"%@ didFailToLoadAd: %@", self.TAG, msg);
    [self updateUI:NO message:[NSString stringWithFormat:NSLocalizedString(@"load_failed", @""), msg]];
}

- (void)didDisplayAd:(MAAd *)ad {
    NSLog(@"%@ didDisplay", self.TAG);
}

- (void)didHideAd:(MAAd *)ad {
    NSLog(@"%@ didHide", self.TAG);
}

- (void)didClickAd:(MAAd *)ad {
    NSLog(@"%@ didClick", self.TAG);
}

- (void)didFailToDisplayAd:(MAAd *)ad withError:(MAError *)error {
    NSLog(@"%@ didFail:%ld %@", self.TAG, (long)error.code, error.description);
}

- (void)didRewardUserForAd:(MAAd *)ad withReward:(MAReward *)reward {
    NSLog(@"%@ didRewardUser", self.TAG);
}

@end

File: AlxAdsDemo/UI/Max/MaxRewardVideoVC.swift
//
//  MaxRewardVideoVC.swift
//  AdsDemo
//
//  Created by liu weile on 2023/6/2.
//

import UIKit
import AppLovinSDK

class MaxRewardVideoVC: BaseUIViewController {
    
    private let TAG = "Max-rewardVideo:"

    private var label:UILabel!

    private var isLoading:Bool=false

    private var rewardAd:MARewardedAd?

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = .white
        navigationItem.title=NSLocalizedString("max_rewardVideo", comment: "")

        let bnLoad=createButton(title: NSLocalizedString("load_ad", comment: ""),  action: #selector(buttonLoad))
        view.addSubview(bnLoad)

        let bnShow=createButton(title: NSLocalizedString("show_ad", comment: ""),  action: #selector(buttonShow))
        view.addSubview(bnShow)

        label=createLabel()
        view.addSubview(label)

        NSLayoutConstraint.activate([
            bnLoad.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnLoad.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnLoad.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnLoad.heightAnchor.constraint(equalToConstant: 50),
            bnLoad.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),

            bnShow.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnShow.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnShow.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnShow.heightAnchor.constraint(equalToConstant: 50),
            bnShow.topAnchor.constraint(equalTo: bnLoad.bottomAnchor, constant: 20),

            label.leftAnchor.constraint(equalTo: view.leftAnchor),
            label.rightAnchor.constraint(equalTo: view.rightAnchor),
            label.widthAnchor.constraint(equalTo: view.widthAnchor),
            label.heightAnchor.constraint(equalToConstant: 50),
            label.topAnchor.constraint(equalTo: bnShow.bottomAnchor, constant: 20),
           
        ])
    }
    
    @objc func buttonLoad(){
        print("bnLoad")
        if isLoading {
            return
        }

        updateUI(true, NSLocalizedString("loading", comment: ""))

        loadAd()
    }
    
    func updateUI(_ loading:Bool,_ msg:String){
        self.isLoading=loading
        self.label.text=msg
    }

    
    @objc func buttonShow(){
        if let ad=rewardAd,ad.isReady {
            ad.show()
        }else{
            print("Ad wasn't ready")
        }
    }
    
    func loadAd() {
        rewardAd=MARewardedAd.shared(withAdUnitIdentifier: AdConfig.Max_Reward_Video_Ad_Id)
        rewardAd?.delegate=self
        rewardAd?.load()
    }

}

extension MaxRewardVideoVC:MARewardedAdDelegate{
    
    func didLoad(_ ad: MAAd) {
        print("\(TAG) didLoad")
        updateUI(false, NSLocalizedString("load_success", comment: ""))
    }
    
    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: MAError) {
        let msg="\(error.code): \(error.description)"
        print("\(TAG) didFailToLoadAd: \(msg)")
        updateUI(false, String(format: NSLocalizedString("load_failed", comment: ""), msg))
    }
    
    func didDisplay(_ ad: MAAd) {
        print("\(TAG) didDisplay")
    }
    
    func didHide(_ ad: MAAd) {
        print("\(TAG) didHide")
    }
    
    func didClick(_ ad: MAAd) {
        print("\(TAG) didClick")
    }
    
    func didFail(toDisplay ad: MAAd, withError error: MAError) {
        print("\(TAG) didFail:\(error.code) \(error.description)")
    }
    
    func didRewardUser(for ad: MAAd, with reward: MAReward) {
        print("\(TAG) didRewardUser")
    }
    
    
}
